server: test flake in TestOAuth2CodeFlow

Looks like our rounding is messed up.

```
	server_test.go:341: verify id token and oauth2 token expiry: expected expired_in to be 2016-11-18 21:16:13 +0000 UTC, got 2016-11-18 21:16:13.506259379 +0000 UTC
```

confirmed

```
$ go get golang.org/x/tools/cmd/stress
$ go test -o server.test github.com/coreos/dex/server
ok  	github.com/coreos/dex/server	0.073s
$ stress ./server.test -test.run=TestOAuth2CodeFlow

/tmp/go-stress496656548
2016/11/18 13:43:26 keys expired, rotating
2016/11/18 13:43:26 keys rotated, next rotation: 2116-10-25 13:43:26 -0800 PST
2016/11/18 13:43:26 keys expired, rotating
2016/11/18 13:43:26 keys rotated, next rotation: 2116-10-25 13:43:26 -0800 PST
2016/11/18 13:43:26 keys expired, rotating
2016/11/18 13:43:26 keys rotated, next rotation: 2116-10-25 13:43:26 -0800 PST
2016/11/18 13:43:26 keys expired, rotating
2016/11/18 13:43:26 keys rotated, next rotation: 2116-10-25 13:43:26 -0800 PST
2016/11/18 13:43:26 keys expired, rotating
2016/11/18 13:43:26 keys rotated, next rotation: 2116-10-25 13:43:26 -0800 PST
2016/11/18 13:43:26 keys expired, rotating
2016/11/18 13:43:26 keys rotated, next rotation: 2116-10-25 13:43:26 -0800 PST
--- FAIL: TestOAuth2CodeFlow (0.09s)
	server_test.go:341: verify id token and oauth2 token expiry: expected expired_in to be 2016-11-18 13:43:56 -0800 PST, got 2016-11-18 13:43:56.510715247 -0800 PST
FAIL
```

--- a/server/server_test.go
+++ b/server/server_test.go
@@ -137,7 +137,7 @@ func TestOAuth2CodeFlow(t *testing.T) {
 	clientSecret := "testclientsecret"
 	requestedScopes := []string{oidc.ScopeOpenID, "email", "offline_access"}
 
-	t0 := time.Now().Round(time.Second)
+	t0 := time.Now()
 
 	// Always have the time function used by the server return the same time so
 	// we can predict expected values of "expires_in" fields exactly.
@@ -171,7 +171,11 @@ func TestOAuth2CodeFlow(t *testing.T) {
 			handleToken: func(ctx context.Context, p *oidc.Provider, config *oauth2.Config, token *oauth2.Token) error {
 				expectedExpiry := now().Add(idTokensValidFor)
 
-				if !token.Expiry.Round(time.Second).Equal(expectedExpiry) {
+				timeEq := func(t1, t2 time.Time, within time.Duration) bool {
+					return t1.Sub(t2) < within
+				}
+
+				if !timeEq(token.Expiry, expectedExpiry, time.Second) {
 					return fmt.Errorf("expected expired_in to be %s, got %s", expectedExpiry, token.Expiry)
 				}
 
@@ -183,7 +187,7 @@ func TestOAuth2CodeFlow(t *testing.T) {
 				if err != nil {
 					return fmt.Errorf("failed to verify id token: %v", err)
 				}
-				if !idToken.Expiry.Round(time.Second).Equal(expectedExpiry) {
+				if !timeEq(idToken.Expiry, expectedExpiry, time.Second) {
 					return fmt.Errorf("expected id token expiry to be %s, got %s", expectedExpiry, token.Expiry)
 				}
 				return nil

